### PR TITLE
serviceaccount: add unit tests and improvements

### PR DESCRIPTION
**Pull request by sebrandon1**
Changes:
- `apiClient` becomes an interface.
- Adjust returns in Create().
- Add unit tests to cover the whole file.

### CODE REVIEW

File: pkg/serviceaccount/serviceaccount_test.go [hunk 1]
```diff
@@ -1 +1,244 @@
...
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestPull(t *testing.T) {
```
Comment (kononovn) [inline pkg/serviceaccount/serviceaccount_test.go:14]: Please add TestServiceAccount* to all test functions

**Review by kononovn — CHANGES_REQUESTED** [review]
1 fix needed.

File: pkg/serviceaccount/serviceaccount_test.go (hunk 1 above)
Comment (sebrandon1) [reply pkg/serviceaccount/serviceaccount_test.go]: Done!

File: pkg/serviceaccount/serviceaccount_test.go [hunk 2]
```diff
@@ -1 +1,244 @@
...
+	"testing"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+)
+
```
Comment (kononovn) [inline pkg/serviceaccount/serviceaccount_test.go:13]: Please add test cases for:

- NewBuilder
- Exist

**Review by kononovn — CHANGES_REQUESTED** [review]
Please add 2 more test cases

File: pkg/serviceaccount/serviceaccount_test.go (hunk 2 above)
Comment (sebrandon1) [reply pkg/serviceaccount/serviceaccount_test.go:13]: Done.

File: pkg/serviceaccount/serviceaccount_test.go [hunk 3]
```diff
@@ -1 +1,328 @@
...
+	testBuilder := buildValidTestBuilder()
+
+	testBuilder.WithOptions(func(builder *Builder) (*Builder, error) {
+		return builder, nil
+	})
+
+	assert.Equal(t, "", testBuilder.errorMsg)
+}
+
+func buildValidTestBuilder() *Builder {
```
Comment (kononovn) [inline pkg/serviceaccount/serviceaccount_test.go:226]: Please move private function to the end of the file

File: pkg/serviceaccount/serviceaccount_test.go [hunk 4]
```diff
@@ -1 +1,328 @@
...
+			runtimeObjects = append(runtimeObjects, testSA)
+		}
+
+		testBuilder := buildTestBuilderWithFakeObjects(runtimeObjects, testSA.Name, testSA.Namespace)
+		err := testBuilder.Delete()
+		assert.Nil(t, err)
+	}
+}
+
+func TestServiceAccountValidate(t *testing.T) {
```
Comment (kononovn) [inline pkg/serviceaccount/serviceaccount_test.go:156]: validate is private function. Please move to the end of the file

File: pkg/serviceaccount/serviceaccount_test.go [hunk 5]
```diff
@@ -1 +1,328 @@
...
+	fakeClient := k8sfake.NewSimpleClientset()
+
+	return NewBuilder(&clients.Settings{
+		K8sClient:       fakeClient,
+		AppsV1Interface: fakeClient.AppsV1(),
+		CoreV1Interface: fakeClient.CoreV1(),
+	}, "test-sa", "test-ns")
+}
+
+func TestNewBuilder(t *testing.T) {
```
Comment (kononovn) [inline pkg/serviceaccount/serviceaccount_test.go:236]: Please move this function as the first on the file.

**Review by kononovn — CHANGES_REQUESTED** [review]
Few formatting issues

File: pkg/serviceaccount/serviceaccount_test.go (hunk 3 above)
Comment (sebrandon1) [reply pkg/serviceaccount/serviceaccount_test.go:320]: Done.

**Pull request merged**